js/indicators/trendline.src.js

```diff
@@ -0,0 +1,128 @@
+/* *
+ *
+ *  License: www.highcharts.com/license
+ *
+ * */
+
+'use strict';
+
+import H from '../parts/Globals.js';
+
+import U from '../parts/Utilities.js';
+var isArray = U.isArray;
+
+var seriesType = H.seriesType;
+
+/**
+ * The Trend line series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.trendline
+ *
+ * @augments Highcharts.Series
+ */
+seriesType(
+    'trendline',
+    'sma',
+    /**
+     * Trendline (linear regression) fits a straight line to the selected data
+     * using a method called the Sum Of Least Squares. This series requires the
+     * `linkedTo` option to be set.
+     *
+     * @sample stock/indicators/trendline
+     *         Exponential moving average indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        7.1.3
+     * @product      highstock
+     * @optionparent plotOptions.trendline
+     */
+    {
+        name: 'TRENDLINE',
```

Comment:
Instead of fixed `name`, use `proto.nameBase`. I think capitals are not needed too, simple `Trendline` should be fine.